File: projecteuler/q4.c
//the palindrome number lies between 10000 and 998001
//so find the number from 998001 which is a palindrome
//find if there is a more efficient way

#include<stdio.h>
#include<stdlib.h>
int palindrome(int i){
	int r;
	int t = i;
	while (t != 0)
  	{
    		r = r * 10;
    		r = r + t%10;
    		t = t/10;
  	}
	if(r==i)
		return 1;
	else
		return 0;
	
}	
int main()
{	int i,count = 0;
	
	for(i=998001;i>10000;i--)
		if(palindrome(i)==1)
		{
			printf("%d",palindrome(i));
			for(int j=999;i>100;i++)
			{
				if(i%j==0)
				{
					printf("%d",i);
					return 0;
				}
				else
					printf("dfg");
			}
		}
	return 1;
}

File: Python-Solutions/q4.py
#palindrome lies between 10000 and 998001
def palindrome(s):
    rev = 0
    i = s
    while(i!=0):
        rev = rev*10+i%10
        i/=10
    if(rev == s):
        return 1
    else:
        return 0

def multiple(n):
    return [x for x in range(100, 1000) if n % x == 0]

for i in range(10000,998002):
    if(palindrome(i)):
        for j in multiple(i):
            if((i/j)<1000 and (i/j)>=99):
                print(i)
        

#print(multiple(4))
#print(palindrome(10001))
    
